libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java

```diff
@@ -299,10 +300,9 @@ static NavigationIndices increaseIndex(RouteProgress routeProgress,
                                                                              List<StepIntersection> intersections) {
     List<Pair<StepIntersection, Double>> distancesToIntersections = new ArrayList<>();
     List<StepIntersection> stepIntersections = new ArrayList<>(intersections);
```

Comment:
Why are we creating `stepIntersections` here? We could check if `intersections` is empty (instead of `stepIntersections.isEmpty()`) and define it down (after the guard clause), close to where it's used. Am I missing something?